app/src/main/java/org/akvo/flow/presentation/navigation/UserOptionsDialog.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.presentation.navigation;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.DialogFragment;
+
+import org.akvo.flow.R;
+import org.akvo.flow.util.ConstantUtil;
+
+public class UserOptionsDialog extends DialogFragment {
+
+    public static final String TAG = "UserOptionsDialog";
+
+    private ViewUser viewUser;
+    private UserOptionListener listener;
+
+    public UserOptionsDialog() {
+    }
+
+    public static UserOptionsDialog newInstance(ViewUser viewUser) {
+        UserOptionsDialog fragment = new UserOptionsDialog();
+        Bundle bundle = new Bundle();
+        bundle.putParcelable(ConstantUtil.VIEW_USER_EXTRA, viewUser);
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        viewUser = getArguments().getParcelable(ConstantUtil.VIEW_USER_EXTRA);
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        listener = (UserOptionListener) getActivity();
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        listener = null;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(final Bundle savedInstanceState) {
+        super.onCreateDialog(savedInstanceState);
+        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+        builder.setTitle(viewUser.getName())
+                .setItems(R.array.user_options, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (which == 0) {
```

Comment:
It could make things more clear if `which` had a more descriptive name and the `0` was a descriptive constant definition.

Reply:
I have fixed this issue